app/src/main/java/com/checkin/app/checkin/utility/Utils.java

```diff
@@ -298,6 +297,27 @@ public static String formatTimeDuration(long milliSec) {
         return "0 Sec";
     }
 
+    public static String formatTimeTaken(long milliSec) {
```

Comment:
I said, this logic is already implemented. You can see it in the method down, **formatDueTime**. You can see how it's used. Maybe you can improve the logic to separate the function to work with time difference.